smmregrid/regrid.py

```diff
@@ -585,9 +598,30 @@ def apply_weights(self, source_data, weights, weights_matrix=None,
         # Clean CDI gridtype (which can lead to issues with CDO interpretation)
         target_da.attrs.pop('CDI_grid_type', None)
 
-
         return target_da
 
+    def _check_nan_variation(self, source_grid, gridtype):
```

Comment:
hmmm, you have a method called `_check_nan_variation` and one `nan_variation_check` ... not really optimal for clarity, can be confusing